File: src/01-easy/power-of-thor-episode-1/solution.ml
let () =
  let line = input_line stdin in
  let parts = String.split_on_char ' ' line in
  let nums = List.map int_of_string parts in
  let lx = List.nth nums 0 and ly = List.nth nums 1 in
  let tx = ref (List.nth nums 2) and ty = ref (List.nth nums 3) in
  while true do
    ignore (input_line stdin);
    let dir = Buffer.create 2 in
    if !ty > ly then (Buffer.add_char dir 'N'; decr ty)
    else if !ty < ly then (Buffer.add_char dir 'S'; incr ty);
    if !tx > lx then (Buffer.add_char dir 'W'; decr tx)
    else if !tx < lx then (Buffer.add_char dir 'E'; incr tx);
    print_endline (Buffer.contents dir)
  done

File: src/01-easy/power-of-thor-episode-1/solution.php
<?php
fscanf(STDIN, "%d %d %d %d", $lx, $ly, $tx, $ty);
while (true) {
    fscanf(STDIN, "%d", $e);
    $dir = "";
    if ($ty > $ly) { $dir .= "N"; $ty--; }
    elseif ($ty < $ly) { $dir .= "S"; $ty++; }
    if ($tx > $lx) { $dir .= "W"; $tx--; }
    elseif ($tx < $lx) { $dir .= "E"; $tx++; }
    echo $dir . "\n";
}
